sds_data_manager/lambda_code/SDSCode/path_helper.py

```diff
@@ -0,0 +1,186 @@
+"""Functions to store file pattern, validate and construct upload path
+"""
+import re
+from dataclasses import dataclass, field
+from datetime import datetime
+
+
+@dataclass
+class FilenamePatternConfig:
```

Comment:
I sort of worry we are going too class-heavy on IMAP... Is there a reason we need this stored/instantiated as an object versus using chains of functions? I think you could chain a bunch of functions together and it would be easier to re-use for other purposes too then.

My personal preference is to use global constants for your valid options and then check against those:
```python
VALID_INSTRUMENTS = {"codice", "glows", ...}
# Could make this more robust later and restrict it to specific data levels for each instrument
VALID_DATALEVELS = {"l0", ...}

def is_filename_valid(filename):
    instrument, datalevel, ... = filename.split("_")
    if instrument not in VALID_INSTRUMENTS:
        raise ValueError(f"{instrument} is not a valid instrument on IMAP, must be one of {VALID_INSTRUMENTS})
```

You could also break those checks out into individual functions if you wanted too.

Reply:
I like the class, personally. That being said, I would probably prefer to pass it in to the functions that use it as a parameter so other users could override it easily. (eg in line 114, maybe you could have `file_pattern_config` as an optional keyword and default to this FilePatternConfig class.)

Reply:
I push new code with these changes. I hope I address your comments.